core/dragged_connection_manager.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2017 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview
+ * @author fenichel@google.com (Rachel Fenichel)
+ */
+'use strict';
+
+goog.provide('Blockly.DraggedConnectionManager');
+
+goog.require('Blockly.RenderedConnection');
+
+/**
+ * TODO: Doc
+ * @constructor
+ */
+Blockly.DraggedConnectionManager = function(block) {
+  this.topBlock_ = block;
+
+  Blockly.selected = block;
+  this.workspace_ = block.workspace;
+
+  this.availableConnections_ = [];
+  this.availableConnectionsCached_ = false;
+
+  this.allDraggedConnections_ = [];
+
+  this.closestConnection_ = null;
+
+  this.localConnection_ = null;
```

Comment:
It would be nice to separate the fields that will change during the drag and the ones that are expected to be consistent.